### ISSUE TITLE
🏁 Crash report v18.2.0

### ISSUE THREAD

**Issue by peperbol** [itchio/itch#782]
socket hangup

Crash log:

```
TypeError: Cannot read property 'headers' of undefined
    at C:\Users\Pepijn W\AppData\Local\itch\app-18.2.0\resources\app.asar\appsrc\util\ibrew\net.js:28:28
    at done (C:\Users\Pepijn W\AppData\Local\itch\app-18.2.0\resources\app.asar\node_modules\needle\lib\needle.js:377:14)
    at ClientRequest.had_error (C:\Users\Pepijn W\AppData\Local\itch\app-18.2.0\resources\app.asar\node_modules\needle\lib\needle.js:384:5)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at TLSSocket.socketCloseListener (_http_client.js:282:9)
    at emitOne (events.js:101:20)
    at TLSSocket.emit (events.js:188:7)
    at TCP._handle.close [as _onclose] (net.js:492:12)
```

**Comment by peperbol** [comment]
So i just installed the app, got the logon screen. Tried possible cridentials. and got a "socket hangup" error in the interface itsself, (or something like that, not sure anymore)
then it started installing a dependency file got the same error again, followed by this crash report. 
i closed the app since "installing depencecy file 5.24" didn't seem to actually progress. and when i reopen I get "installing depencecy file 5.24" again with no real progress.
Do i reinstall? do you need me to check something 1st before reinstalling?

**Comment by fasterthanlime** [comment]
Relaunching the app should probably fix it — socket hangups are usually a sign of unreliable internet access (are you on wifi? weak signal by any chance?), and it looks like that particular code tries to access the network response even when there's been a network error — so, I'll fix that code, but apart from retrying until the requests succeed, there's not much else for you to do!

(No need to reinstall at all though, there's no signs the install is corrupted)

**Comment by fasterthanlime** [comment]
Well https://github.com/itchio/itch/commit/d226d52431bdd3f4cb0e69dc157e4d11efd0cf1a shipped with 18.3.0, closing!
